src/MongoDB.Driver.Core/ChangeStreamDocument.cs

```diff
@@ -75,7 +75,21 @@ public sealed class ChangeStreamDocument<TDocument> : BsonDocumentBackedClass
         /// <value>
         /// The full document.
         /// </value>
-        public TDocument FullDocument => GetValue<TDocument>(nameof(FullDocument), default(TDocument));
+        public TDocument FullDocument
+        {
+            get
+            {
+                // if TDocument is BsonDocument avoid deserializing it again to prevent possible duplicate element name errors
+                if (typeof(TDocument) == typeof(BsonDocument) && BackingDocument.TryGetValue("fullDocument", out var fullDocument))
+                {
+                    return (TDocument)(object)fullDocument.AsBsonDocument;
+                }
+                else
+                {
+                    return GetValue<TDocument>(nameof(FullDocument), default(TDocument));
+                }
+            }
```

Comment:
This change returns an already deserialized BsonDocument (possibly containing duplicate element names) to be returned directly without a second deserialization which will throw an exception if any duplicate element names are present.

Reply:
My suggestion requires changes in two new classes(BsonSerializationInfo.cs and BsonDocumentBackedClassSerializer.cs). Your way requires changes in one new class (ChangeStreamDocument.cs). So, it's not too big code difference..
Also, with my suggestion, we delegate all configuration work to the `ChangeStreamDocumentSerializer.cs` which is the natural place to configure this stuff. In your way, we do it in two different places.

Since the difference between these two ways is really small, I would still consider the implementation of this change unless I missed something

Reply:
I want to stick with the approach currently in my branch because:

1. It is simpler
2. The changes are localized to two ChangeStreamDocument related classes
3. No public APIs are changed
4. It avoids double deserialization of the FullDocument

Reply:
>It is simpler

several lines less that in the suggestion..

>The changes are localized to two ChangeStreamDocument related classes

but the ticket logic is not localized in the changeStream serializer, instead we make the same change in two different classes

>No public APIs are changed

I don't see how it can be a problem, we're just adding one way to configure deserialization. We even not create it, we just lift up already created functionality to the change stream level.

>It avoids double deserialization of the FullDocument

not sure what you mean, but in any case, this is how it was designed. And we have a number of similar cases when we use BsonDocument serializer for the field.

And we still lose the designed functionality which we configure via `RegisterMember`.

Reply:
However, I will approve this PR. Maybe we can live with this change.